app/services/joke-service.server.ts

```diff
@@ -61,7 +63,23 @@ export class JokeService extends BaseService {
       where: { queued: true },
       take: 15,
       orderBy: { queuedAt: "desc" },
-      include: { submitter: { select: { displayName: true, profileImgUrl: true } } },
+      include: {
+        submitter: { select: { displayName: true, profileImgUrl: true } },
+        ratings: {
+          select: { score: true, userId: true },
+          where: { userId: this.user.id },
+        },
+      },
+    });
+
+    const ratings = await db.rating.groupBy({
+      by: ["score", "jokeId"],
+      where: {
+        jokeId: {
+          in: jokes.map((joke) => joke.id),
+        },
+      },
```

Comment:
I wasn't able to get this into a single query. 😠 close enough for what we're doing